vmm/src/seccomp_filters.rs

```diff
@@ -0,0 +1,295 @@
+            allow_syscall(libc::SYS_read),
+            allow_syscall(libc::SYS_readlink),
+            allow_syscall(libc::SYS_recvfrom),
+            allow_syscall(libc::SYS_recvmsg),
+            allow_syscall(libc::SYS_restart_syscall),
+            allow_syscall(libc::SYS_rt_sigaction),
+            allow_syscall(libc::SYS_rt_sigprocmask),
+            allow_syscall(libc::SYS_rt_sigreturn),
+            allow_syscall(libc::SYS_sched_getaffinity),
+            allow_syscall(libc::SYS_sendmsg),
+            allow_syscall(libc::SYS_set_robust_list),
+            allow_syscall(libc::SYS_set_tid_address),
+            allow_syscall(libc::SYS_sigaltstack),
+            allow_syscall_if(
+                libc::SYS_socket,
+                or![
+                    and![Cond::new(0, ArgLen::DWORD, Eq, libc::AF_UNIX as u64)?],
+                    and![Cond::new(0, ArgLen::DWORD, Eq, libc::AF_INET as u64)?],
+                ],
+            ),
+            allow_syscall(libc::SYS_socketpair),
+            allow_syscall(libc::SYS_stat),
+            allow_syscall(libc::SYS_statx),
+            allow_syscall(libc::SYS_tgkill),
+            allow_syscall(libc::SYS_unlink),
+            allow_syscall(libc::SYS_wait4),
+            allow_syscall(libc::SYS_write),
+        ]
+        .into_iter()
+        .collect(),
+        SeccompAction::Trap,
+    )?)
+}
+
+/// The filter containing the white listed syscall rules required by the API to
+/// function.
+pub fn api_thread_filter() -> Result<SeccompFilter, Error> {
+    Ok(SeccompFilter::new(
+        vec![
+            allow_syscall(libc::SYS_accept4),
+            allow_syscall(libc::SYS_bind),
+            allow_syscall(libc::SYS_close),
+            allow_syscall(libc::SYS_dup),
+            allow_syscall(libc::SYS_epoll_create1),
+            allow_syscall(libc::SYS_epoll_ctl),
+            allow_syscall(libc::SYS_epoll_wait),
+            allow_syscall(libc::SYS_exit),
+            allow_syscall(libc::SYS_futex),
+            allow_syscall(libc::SYS_getrandom),
+            allow_syscall_if(libc::SYS_ioctl, create_api_ioctl_seccomp_rule()?),
+            allow_syscall(libc::SYS_listen),
+            allow_syscall(libc::SYS_madvise),
+            allow_syscall(libc::SYS_munmap),
+            allow_syscall(libc::SYS_recvfrom),
+            allow_syscall(libc::SYS_sigaltstack),
+            allow_syscall(libc::SYS_socket),
+            allow_syscall(libc::SYS_write),
+        ]
+        .into_iter()
+        .collect(),
+        SeccompAction::Trap,
+    )?)
+}
+
+/// Generate a BPF program based on a seccomp level value.
+pub fn get_seccomp_filter(
+    seccomp_level: &SeccompLevel,
+    thread_type: Thread,
+) -> Result<BpfProgram, SeccompError> {
+    let filter = match thread_type {
+        Thread::Vmm => vmm_thread_filter(),
+        Thread::Api => api_thread_filter(),
+    };
+    match *seccomp_level {
+        SeccompLevel::None => Ok(vec![]),
+        SeccompLevel::Basic => filter
```

Comment:
What does the `Basic` filter do?

Reply:
It does not take into account the conditional syscalls. Instead it simply whitelist the entire syscall, no matter which parameter is passed to it.
I don't think we would ever want to use it.

Reply:
So it could be deleted then?

Reply:
Yes I can remove this :)